### ISSUE TITLE
[BUG] CustomLearningViewer.aspx shows a blank page for all users

### ISSUE THREAD

**Issue by eariel** [pnp/custom-learning-office-365#248]
We have been using this tool for a while with no issues. But today we have noticed that the viewer page is not showing any contents. 
When I edit the page, I can see the webpart options, and select the scenario > click on Save Button (the content is still visible) but when I refresh the page, is blank again.

The page is not showing content of any playlist (custom or predefined).

The app version is 3.0.3.0 and we have no clue what is causing this issue. 

With F12 Dev Tool in Chrome I've found this:

"Access to XMLHttpRequest at 'https://waconatm.officeapps.live.com/apc/trans.gif?14d53d4e9aa99866405d0c4f36f0f4ad' from origin 'https://mydomain.sharepoint.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource."

But, to be honest, I don't know if this error was present before this happen.

Does anyone have the similar issue with this?

Thanks in advance.

**Comment by juliemturner** [comment]
Ok, so... that error would have nothing to do with learning pathways, you're looking for an error related to that specifically... but, you can start with just what are all the errors on the page.  Regardless there is an updated web part (3.0.4) so I would get that installed, and then if you can get me a screenshot or listing or any other errors that would be helpful.

**Comment by eariel** [comment]
Hi Julie, I'll explain step by step what I did I case this helps the community.

1º Uptate to version 3.0.4 -> not working
2º Create a new playlist -> not working
3º Check if the playlist is working outside page CustomLearningViewer.aspx -> playlist are working if I don't use that page
4º Edit CustomLearningViewer.aspx and select a different playlist. I realized that I need to provide metadata (Technology, Level, Audience) to the playlist to make it visible. -> The page is working now.
5º Edit one of the playlist and add metadata (Technology, Level and Audience) and it's working too.
6º Edit the playlist and remove the metadata (Technology, level and Audiencie) and the page is still working.

I guess that some configuration element is updating when I edit this metadata, and after this update, the modified playlist is working. These data are not mandatory, but I had to do this in every playlist to be able to show them on CustomLearningViewer.aspx.

Thanks for your help.

**Comment by juliemturner** [comment]
They are not mandatory. We have had several other people mention that playlists without the Technology weren't showing up (never heard that audience and level affected things) but I cannot get anyone willing to work with me long enough to fix this issue, as I cannot replicate it. All I've ever seen is that once the user realizes that they need to refresh the page (on the viewer web part) after they've updated the playlist in the admin web part they report they're no longer having an issue. Based on the fact that you've "fixed" them all already I suspect you won't be able to replicate the issue either. Happy it's working now.

**Comment by heathherd** [comment]
Hello, 

I am experiencing the similar issue.. I followed the instructions modifying the json file with my tenant information but when the app is installed each tab is blank.

**Comment by juliemturner** [comment]
I've never shared any information about "modifying a json file" can you please provide additional information?

**Comment by juliemturner** [comment]
Also, in this case it looks like you have a different issue so please open a new issue so that we can track is separately.

**Comment by heathherd** [comment]
Here is what I'm following for editing the json manafest
https://github.com/msft-teams/tools/blob/master/getstartedapp/installandconfig.md

**Comment by juliemturner** [comment]
Yes, you indeed have your own issue, which is unrelated to the one this issue is about.  But further that issue isn't related specifically to learning pathways, but instead is an augmentation of learning pathways related to using it in Teams.  I should probably full-stop there and tell you that you should submit your issue to the repo in which you pointed to, but, I should also tell you that it will be difficult for them to help you if you don't have learning pathways successfully running in your SharePoint site.  So I would suggest you verify that first and then reach out by submitting an issue to that Teams repo.  If you are having difficulty getting learning pathways running in SharePoint please feel free to open up a new issue in this repo to get help. Please try and be as thorough as you can explaining where you're at and what exactly you're experiencing.

**Comment by heathherd** [comment]
I figured out the issue, thanks

**Comment by GHVandeput** [comment]
@heathherd How did you solve the isssue?
